src/light_data.c

```diff
@@ -356,22 +348,144 @@ long light_is_light_allocated(long lgt_id)
 
 void light_set_light_position(long lgt_id, struct Coord3d *pos)
 {
-  _DK_light_set_light_position(lgt_id, pos);
+  // _DK_light_set_light_position(lgt_id, pos);
+  struct Light *lgt = &game.lish.lights[lgt_id];
+  if ( *(unsigned short *)&lgt->mappos.x.stl != *(unsigned short *)&pos->x.stl
+    || *(unsigned short *)&pos->y.stl != *(unsigned short *)&lgt->mappos.y.stl
+    || *(unsigned short *)&pos->z.stl != *(unsigned short *)&lgt->mappos.z.stl )
+  {
+    if ( !(lgt->flags & LgtF_Dynamic) )
+    {
+      stat_light_needs_updating = 1;
+      unsigned char range = lgt->range;
+      long end_y = lgt->mappos.y.stl.num + range;
+      long end_x = lgt->mappos.x.stl.num + range;
+      if ( end_y > 255 )
+      {
+        end_y = 255;
+      }
+      if ( end_x > 255 )
+      {
+        end_x = 255;
+      }
+      long beg_y = lgt->mappos.y.stl.num - range;
+      if ( beg_y < 0 )
+      {
+        beg_y = 0;
+      }
+      long beg_x = lgt->mappos.x.stl.num - range;
+      if ( beg_x < 0 )
+      {
+        beg_x = 0;
+      }
+      light_signal_stat_light_update_in_area(beg_x, beg_y, end_x, end_y);
+    }
+    *(unsigned long *)&lgt->mappos.x.stl = *(unsigned long *)&pos->x.stl;
+    *(unsigned short *)&lgt->mappos.z.stl = *(unsigned short *)&pos->z.stl;
+    lgt->flags |= LgtF_Unkn08;
+  }
 }
 
 void light_remove_light_from_list(struct Light *lgt, struct StructureList *list)
 {
-  _DK_light_remove_light_from_list(lgt, list);
+  // _DK_light_remove_light_from_list(lgt, list);
+  TbBool Removed = false;
+  struct Light *lgt2;
+  struct Light *i;
+  if ( lgt->field_1 & 1 )
```

Comment:
Enum

Reply:
Enum to what?

Reply:
I think it is `LgtF_Allocated` instead of `1`

Reply:
field_1 is not flags though.

Reply:
Ouch. But it used like flag or what it means actually?

Reply:
I don't know what field_1 is I'm afraid. Judging by the fact it uses and operations, it could be some sort of flag field, but apparently it's not the same as flags.

Reply:
call it "flag_2" )